### DISCUSSION TITLE
KafkaMirrorMaker2 - Multiple mirrors

### DISCUSSION THREAD

**Discussion by lenin-joseph1** [strimzi/strimzi-kafka-operator#7795] [Q&A]
Hi, 

We have an use case where we want to replicate "msk.*" topics with cluster alias as a prefix, and replicate "_schemas" without any prefix using the same MirrorMaker instance, so we have used two mirrors as below, could you advice if this is the right approach?

---
apiVersion: kafka.strimzi.io/v1beta2
kind: KafkaMirrorMaker2
metadata:
  name: mm2-cluster
  namespace: kafka-ns
spec:
  version: 3.0.0
  replicas: 2
  connectCluster: target-kafka
  clusters:
  - alias: source-kafka
  - alias: target-kafka
  mirrors:
  - sourceCluster: source-kafka
    targetCluster: target-kafka
    sourceConnector:
      config:
        replication.factor: 3
        offset-syncs.topic.replication.factor: 3
        tasks.max: 6
    heartbeatConnector:
      config:
        heartbeats.topic.replication.factor: 3
    checkpointConnector:
      config:
        checkpoints.topic.replication.factor: 3
        sync.group.offsets.enabled: true
    topicsPattern: "msk.*"
    groupsPattern: ".*"
  - sourceCluster: source-kafka
    targetCluster: target-kafka
    sourceConnector:
      config:
        replication.factor: 3
        offset-syncs.topic.replication.factor: 3
        tasks.max: 6
        replication.policy.separator: ""
        replication.policy.class: io.strimzi.kafka.connect.mirror.IdentityReplicationPolicy
    heartbeatConnector:
      config:
        heartbeats.topic.replication.factor: 3
    checkpointConnector:
      config:
        checkpoints.topic.replication.factor: 3
        sync.group.offsets.enabled: true
    topicsPattern: "_schemas"
    groupsPattern: ".*"
---

**Comment by yxw** [comment]
@lenin-joseph1, did you managed to archive the goal ?
We had a similar case where we want to have two mirrors for the same source/target cluster, each apply to a different topic pattern. It didn't seem to working as the second would make the first kept on being restarted.
